### PR TITLE
readme: Make README more legible

### PR DESCRIPTION
**Pull request by jazzdan**
Two outstanding TODOs that I'd like feedback on:

1. Should we explain how `start_fast_build` is fast, and what image and container builds are?
2. We should probably explain the concept of the "active" build (if you call `start_fast_build` before calling `stop_build` there will be an error) but I'm not sure the best way to do that.

In the future we should probably use Sphinx, like we did in windmilleng/windmill.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,88 +25,125 @@ def backend():
...
 
-* Args:
-  * `dockerfile_path`: **str**
-  * `img_name`: **str**, e.g. blorgdev/backend or gcr.io/project-name/bucket-name
-  * `entrypoint?`: **str**
-* Returns: **Image**
+```python
+class Repo:
+  def path(path: str) -> localPath:
+    """Returns the absolute path to the file specified at path in the repo.
```
Comment (maiamcc) [inline README.md]: imo `path` should be in backticks in this docstring

File: README.md [hunk 2]
```diff
@@ -22,88 +25,125 @@ def backend():
...
+    Returns:
+      A localPath resource, representing a local path on disk.
+    """
+```
+
+#### start_fast_build
 
-#### add(src, dest)
-Adds the content from `src` into the image at path `dest`. Paths must be relative.
+Initiates a docker image build that uses a cache and supports `add`s and `run`s.
```
Comment (maiamcc) [inline README.md]: Maybe:
```
Initiates a docker image build that supports `add`s and `run`s, and that uses a cache for subsequent builds.
```

File: README.md [hunk 3]
```diff
@@ -22,88 +25,125 @@ def backend():
...
+```python
+def add(src: Union[localPath, Repo], dest: str) -> None
+```
+
+#### run
 
-#### run(cmd, trigger?)
 Runs `cmd` as a build step in the image.
-If the `trigger` file is specified, the build step is only run if the file is changed. Path must be relative.
+If the `trigger` file is specified, the build step is only run if the file is changed.
```
Comment (maiamcc) [inline README.md]: > If the `trigger` arg is specified (a file or list of files), the build step is only on on changes to the given file(s).

File: README.md [hunk 4]
```diff
@@ -22,88 +25,125 @@ def backend():
...
 Creates a composite service; tilt will deploy (and watch) all services returned by the functions in `service_fns`.
 
-* Args:
-  * `service_fns`: array of functions that each return **Service**
-* Returns: **Service**
+```python
+def composite_service(List[Callable[[], Service]]) -> Service
+```
+
+#### Service
```
Comment (maiamcc) [inline README.md]: maybe put this def above `composite_service`?

Comment (maiamcc) [reply README.md]: also aren't we no longe calling these "services"?

File: README.md [hunk 5]
```diff
@@ -22,88 +25,125 @@ def backend():
...
+#### read_file
 
-#### read_file(file_path)
 Reads file and returns its contents.
 
-* Args:
-  * `file_path`: **str**
-* Returns: **str**
+```python
+def read_file(file_path: st) -> str
```
Comment (maiamcc) [inline README.md]: s/st/str

File: README.md (hunk 4 above)
Comment (jazzdan) [reply README.md]: Yeah it wasn't clear to me whether that was an internal rebranding our an external one as well. Happy to go through and make that change in a subsequent PR.

**Pull request merged**